backend/src/main/java/atlantbh/restaurants/models/filters/RestaurantFilterBuilder.java

```diff
@@ -39,26 +42,30 @@ public Criteria addConditions(Criteria rootCriteria, boolean isCountCriteria) {
             rootCriteria.createAlias("cousine", "cous");
             rootCriteria.createAlias("location", "loc");
             rootCriteria.add(criterion);
-        }
-        if (StringUtils.isNotBlank(name)) {
-            rootCriteria.add(Restrictions.ilike("name", name, MatchMode.ANYWHERE));
-        }
-        if (StringUtils.isNotBlank(category)) {
-            rootCriteria.createAlias("category", "cat");
-            rootCriteria.add(Restrictions.ilike("cat.name", category, MatchMode.ANYWHERE));
-        }
-        if (priceRange != null && priceRange > 0) {
-            rootCriteria.add(Restrictions.eq("priceRange", priceRange));
-        }
-        if (StringUtils.isNotBlank(location)) {
-            rootCriteria.createAlias("location", "loc");
-            Criterion criterion = Restrictions.or(Restrictions.ilike("loc.city", location, MatchMode.ANYWHERE),
-                    Restrictions.ilike("loc.country", location, MatchMode.ANYWHERE));
-            rootCriteria.add(criterion);
-        }
-        if (StringUtils.isNotBlank(cousine)) {
-            rootCriteria.createAlias("cousine", "c");
-            rootCriteria.add(Restrictions.ilike("c.name", cousine, MatchMode.ANYWHERE));
+        } else { // otherwise creating aliases would throw exception
```

Comment:
Now query can not be in combination with other properties for filtering. Remove if.. Create aliases before you do anything else (it would be slower, but you won't notice).

```
rootCriteria.createAlias("category", "cat");
rootCriteria.createAlias("cousine", "cous");
rootCriteria.createAlias("location", "loc");

if (query)

If (cousine)
```
Other solution would be that you check if alias is already present, if it is, dont create it again..

Reply:
Fixed in 0cd4debab433260ab6a8267aa0b72b31db581edb